### PR TITLE
CSHARP-2661: Show Test Results in Evergreen

### PR DESCRIPTION
**Pull request by durran**
https://spruce.mongodb.com/version/5ff6e948850e615039a261b7/tasks

Legacy UI is better at visualisations:
Screenshot 2021-01-07 at 12 42 56
Screenshot 2021-01-07 at 12 44 54

### CODE REVIEW

**Comment by DmitryLukyanov** [comment]
what data is expected in case of error? I see here more data than I thought it would be: [link](https://evergreen.mongodb.com/test_log/dot_net_driver_tests_snappy_compression__version~latest_os~windows_64_topology~standalone_auth~noauth_ssl~nossl_compressor~snappy_test_netstandard20_patch_3dfeb60f9f8e5a58021d96b2896cf2ff3fbee951_5ff5cfd10ae6065f984806f5_21_01_06_14_57_34/0/MongoDB.Driver.Core.Operations.CreateIndexesOperationTests.MongoDB.Driver.Core.Operations.CreateIndexesOperationTests.Execute_should_work_when_expireAfter_has_value_async__False_#L0) I thought it would be just a single failed test like in java: [java patch](https://evergreen.mongodb.com/test_log/mongo_java_driver_tests_jdk8_unsecure__version~latest_os~linux_topology~replicaset_auth~noauth_ssl~nossl_jdk~jdk8_test_patch_1a568955081e4a690a582293c2c012175c618f82_5fd8db01d6d80a01e1ee5f5c_20_12_15_15_49_36/0/com.mongodb.internal.operation.ChangeStreamOperationProseTestSpecification.should_resume_after_single_getMore_Error__async__false___1_)  (I guess java page shows one test with additional details?)

Not sure whethere it's related to this ticket itself

**Comment by durran** [comment]
@DmitryLukyanov I think the legacy UI is the same as the java driver in this case: https://evergreen.mongodb.com/task/dot_net_driver_tests_snappy_compression__version~latest_os~windows_64_topology~standalone_auth~noauth_ssl~nossl_compressor~snappy_test_netstandard20_patch_3dfeb60f9f8e5a58021d96b2896cf2ff3fbee951_5ff5cfd10ae6065f984806f5_21_01_06_14_57_34

**Comment by durran** [comment]
So I found why it's doing this.

Junit.Xml.TestLogger writes system out and system error to the `<system-out>` and `<system-error>` tags in the test results. This is not configurable. Evergreen when printing the failure message will append system out and system error to the end of the individual failure on each one (https://github.com/evergreen-ci/evergreen/blob/83945041f84a60db700e61e3b1fd09c08860ec37/agent/command/results_xunit.go#L208). So basically in our case we will have the extra output on each failure.

File: tests/AtlasConnectivity.Tests/AtlasConnectivity.Tests.csproj [hunk 1]
```diff
@@ -45,6 +45,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
     <PackageReference Include="xunit" Version="2.4.0" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
+    <PackageReference Include="XunitXml.TestLogger" Version="2.1.26" />
```
Comment (DmitryLukyanov) [inline tests/AtlasConnectivity.Tests/AtlasConnectivity.Tests.csproj]: why somewhere you use `Xunit` and somewhere `Junit`? Should it be just `xunit`?

Comment (durran) [reply tests/AtlasConnectivity.Tests/AtlasConnectivity.Tests.csproj]: The Xunit logger doesn't print out the correct format that Evergreen understands. The Junit logger does.

Comment (durran) [reply tests/AtlasConnectivity.Tests/AtlasConnectivity.Tests.csproj]: (It's a bit misleading as the Evergreen task is attach.xunit but it really means Junit format.)

Comment (DmitryLukyanov) [reply tests/AtlasConnectivity.Tests/AtlasConnectivity.Tests.csproj]: ok (it looks like we cannot do more here), but anyway let's be consistent and add the same logger everywhere

**Review by DmitryLukyanov — COMMENTED** [review]
https://github.com/durran/mongo-csharp-driver/pull/7#discussion_r553336466 it's not answered as I see

**Comment by durran** [comment]
@DmitryLukyanov I fixed the remaining 3 instances of your comment now.

**Review by DmitryLukyanov — APPROVED** [review]
idk details about the whole reason why we use Junit instead Xunit, but I rely on you here, LGTM

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Comment by jyemin** [comment]
@durran I'm going to close this and open up a new PR on the main repo, after rebasing.

**Pull request closed without merging**